transfer-sh.el

```diff
@@ -54,39 +54,69 @@
   :type '(string)
   :group 'transfer-sh)
 
+(defcustom transfer-sh-upload-agent-command
+  (cond
+   ((executable-find "curl")
+    "curl")
+   ((executable-find "wget")
+    "wget"))
+  "Command used to upload files to transfer.sh"
+  :type '(string)
+  :group 'transfer-sh)
+
+(defcustom transfer-sh-upload-agent-arguments
+  (cond
+   ((executable-find "curl")
+    (list "--silent" "--upload-file"))
+   ((executable-find "wget")
+    (list "--method" "PUT" "--output-document" "-"  "--no-verbose" "--quiet" "--body-file")))
+  "Suffix arguments to `transfer-sh-upload-agent-command'"
+  :group 'transfer-sh)
+
 ;;;###autoload
-(defun transfer-sh-upload-file-async (local-filename &optional remote-filename)
+(defun transfer-sh-upload-file-async (local-filename)
```

Comment:
Maybe it is not a good idea to break the API by changing the arguments here.
See next comment.

Reply:
solved